modeshape-persistence-tests/pom.xml

```diff
@@ -0,0 +1,332 @@
+                <!--
+                Connect this plugin to the maven lifecycle around the integration-test phase:
+                start the container in pre-integration-test and stop it in post-integration-test.
+                -->
+                <executions>
+                    <execution>
+                        <id>start</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>build</goal>
+                            <goal>start</goal>
+                        </goals>
+                    </execution>
+                    <execution>
+                        <id>stop</id>
+                        <phase>post-integration-test</phase>
+                        <goals>
+                            <goal>stop</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            
+            
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-failsafe-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>integration-test-mysql</id>
+                        <goals>
+                            <goal>integration-test</goal>
+                        </goals>
+                        <configuration>
+                            <systemPropertyVariables>
+                            <!-- Make these available to the tests via system properties -->
+                                <database.driver>com.mysql.jdbc.Driver</database.driver>
+                                <database.url>jdbc:mysql://localhost:3306/${database.name}</database.url>
+                                <database.user>${database.user}</database.user>
+                                <database.password>${database.password}</database.password>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>integration-test-postgres</id>
+                        <goals>
+                            <goal>integration-test</goal>
+                        </goals>
+                        <configuration>
+                            <systemPropertyVariables>
+                            <!-- Make these available to the tests via system properties -->
+                                <database.driver>org.postgresql.Driver</database.driver>
+                                <database.url>jdbc:postgresql://localhost/${database.name}</database.url>
+                                <database.user>${database.user}</database.user>
+                                <database.password>${database.password}</database.password>
+                            </systemPropertyVariables>
+                        </configuration>
+                    </execution>
+                    <execution>
```

Comment:
Yes, it is required. Failsafe always runs:

* `pre-integration-test` to set up the test infrastructure (which in our case is the Docker containers)
* `integration-test` to run the integration tests using Surefire;
* `post-integration-test` to always tear down the test infrastructure (which in our case is the Docker containers);
* `verify` to check whether any integration tests failed, and if so stops Maven.

This is how Failsafe ensures that the test infrastructure is always torn down before Maven is stopped. See the [Failsafe documentation](http://maven.apache.org/surefire/maven-failsafe-plugin/) for details.